OpenSim/Analyses/JointReaction.cpp

```diff
@@ -567,41 +558,34 @@ record(const SimTK::State& s)
     for(int i=0; i<numOutputJoints; i++) {
         JointReactionKey currentKey = _reactionList[i];
         const Joint& joint = *currentKey.joint;
-        Vec3 force = allForcesVec[currentKey.jointIndex];
-        Vec3 moment = allMomentsVec[currentKey.jointIndex];
-        const PhysicalFrame& expressedInBody = *currentKey.expressedInFrame;
-        
-        // find the point of application of the joint load on the child
-        // in the ground reference frame
-        Vec3 childLocationInGlobal = joint.getChildFrame().getTransformInGround(s_analysis).p();
-        // set the point of application to the joint location in the child body
-        Vec3 pointOfApplication(0,0,0);
+        const Frame& expressedInBody = *currentKey.expressedInFrame;
+        SpatialVec jointReaction;
+        Vec3 pointOfApplication;
         
-        // check if the load on the child needs to be converted to an equivalent
-        // load on the parent body.
+        // check if the load requested is on the parent or child
         if(!currentKey.isAppliedOnChild){
-            /*Take reaction load from child and apply on parent*/
-            force = -force;
-            moment = -moment;
-            Vec3 parentLocationInGlobal = joint.getParentFrame().getTransformInGround(s_analysis).p();
-
-            // define vector from the mobilizer location on the child to the location on the parent
-            Vec3 translation = parentLocationInGlobal - childLocationInGlobal;
-            // find equivalent moment if the load is shifted to the parent location
-            moment -= translation % force;
+            jointReaction = joint.calcReactionOnParentExpressedInGround(s_analysis);
```

Comment:
This isn't really a test. Should `calcReactionOnParentExpressedInGround()` be covered by a test case?